### PR TITLE
refactor: move platform definitions to core

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: .bazelci/presubmit.yml [hunk 1]
```diff
@@ -134,34 +134,34 @@ tasks:
...
     - "--build_tag_filters=-cypress"
     build_targets:
     - "//internal/..."
     test_flags:
-    - "--platforms=@build_bazel_rules_nodejs//toolchains/node:darwin_amd64"
+    - "--platforms=@rules_nodejs//nodejs:darwin_amd64"
     test_targets:
     # //internal/node/test:nodejs_toolchain_darwin_amd64_test is a "manual" test that must be run
-    # explicitly with --platforms=@build_bazel_rules_nodejs//toolchains/node:darwin_amd64 set
+    # explicitly with --platforms=darwin_amd64 set
```
Comment (thesayyn) [inline .bazelci/presubmit.yml:147]:
```suggestion
    # explicitly with --platforms=@rules_nodejs//nodejs:darwin_amd64 set
```

File: .bazelci/presubmit.yml [hunk 2]
```diff
@@ -134,34 +134,34 @@ tasks:
...
     - "--build_tag_filters=-cypress"
     build_targets:
     - "//internal/..."
     test_flags:
-    - "--platforms=@build_bazel_rules_nodejs//toolchains/node:windows_amd64"
+    - "--platforms=@rules_nodejs//nodejs:windows_amd64"
     test_targets:
     # //internal/node/test:nodejs_toolchain_windows_amd64_test is a "manual" test that must be run
-    # explicitly with --platforms=@build_bazel_rules_nodejs//toolchains/node:windows_amd64 set
+    # explicitly with --platforms=windows_amd64 set
```
Comment (thesayyn) [inline .bazelci/presubmit.yml:164]:
```suggestion
    # explicitly with --platforms=@rules_nodejs//nodejs:windows_amd64 set
```

File: .bazelci/presubmit.yml [hunk 3]
```diff
@@ -250,14 +250,14 @@ tasks:
...
+    - "--platforms=@rules_nodejs//nodejs:linux_amd64"
     build_targets:
     - "//internal/..."
     test_flags:
-    - "--platforms=@build_bazel_rules_nodejs//toolchains/node:linux_amd64"
+    - "--platforms=@rules_nodejs//nodejs:linux_amd64"
     test_targets:
     # //internal/node/test:nodejs_toolchain_linux_amd64_test is a "manual" test that must be run
-    # explicitly with --platforms=@build_bazel_rules_nodejs//toolchains/node:linux_amd64 set
+    # explicitly with --platforms=linux_amd64 set
```
Comment (thesayyn) [inline .bazelci/presubmit.yml:260]:
```suggestion
    # explicitly with --platforms=@rules_nodejs//nodejs:linux_amd64 set
```

File: nodejs/BUILD.bazel [hunk 4]
```diff
@@ -27,3 +28,11 @@ toolchain_type(
...
     visibility = ["//visibility:public"],
 )
+
+[
+    platform(
+        name = key,
+        constraint_values = values.compatible_with,
+    )
+    for key, values in PLATFORMS.items()
+]
```
Comment (thesayyn) [inline nodejs/BUILD.bazel:38]: Suggestion/question: wouldn’t it be more self-explanatory if we were to make a `platforms` directory and move this inside the BUILD file?

**Review by thesayyn — COMMENTED** [review]
A few minor changes otherwise LGTM!

File: nodejs/BUILD.bazel (hunk 4 above)
Comment (alexeagle) [reply nodejs/BUILD.bazel:38]: eh, it's a pretty trivial difference. I'd say that the data is already encoded in the graph by `bazel query 'kind(platform, @rules_nodejs//nodejs:all)'` and it's not enough semantics to be worth another BUILD file. It makes all the target names longer and means our distribution contents needs another filegroup. Users barely ever type this (and rules_docker is gaining transitions so you won't need to use `--platforms` there anymore.)

**Pull request merged**